pkg/controller/configmap/config.go

```diff
@@ -623,6 +621,24 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 		if !needChange.agent && inSlice(sec, operatortypes.AgentSections) {
 			needChange.agent = true
 		}
+		if !needChange.agent && inSlice(sec, []string{"k8s", "coe"}) {
+			if hasSection(prevCfg, sec) != hasSection(currCfg, sec) {
+				needChange.agent = true
+			} else {
+				// We can assert that section `sec` exists in both prevCfg and currCfg here.
+				for _, key := range operatortypes.AgentKeys {
+					if prevCfg.Section(sec).HasKey(key) != currCfg.Section(sec).HasKey(key) {
```

Comment:
if changed key is not from AgentKeys, do we need to restart node agent as well? if not, we need to check added key or removed key from AgentKeys or not to decide if restarting node agent or not?

Reply:
> if changed key is not from AgentKeys, do we need to restart node agent as well? if not, we need to check added key or removed key from AgentKeys or not to decide if restarting node agent or not?

Not really get you point, but I think this judgement is to restart node-agent when there is added or removed key?

Reply:
I see your point. So you only want to restart node agent when key is from AgentRestartKeys, and the other key in k8s and coe are be ignored no matter what changes are made.

Reply:
Yes, that's right.